flare_portal/users/decorators.py

```diff
@@ -31,3 +32,34 @@ def _wrapped_view(
         return _wrapped_view
 
     return decorator(function)
+
+
+def has_researcher_access(
+    function: Callable,
+    owner_only: bool = False,
+    login_url: str = None,
+    redirect_field_name: str = REDIRECT_FIELD_NAME,
+) -> Callable:
+    def decorator(view_func: Callable) -> Callable:
+        @wraps(view_func)
+        def _wrapped_view(
+            request: HttpRequest, *args: Any, **kwargs: Any
+        ) -> HttpResponse:
+            if isinstance(request.user, User):
```

Comment:
**Suggestion:** Is this for MyPy? Checking `request.user.is_authenticated` should be enough if you use `django-stubs`.